### PR TITLE
Adding in aXe accessibility testing

### PR DESCRIPTION
**Pull request by Guntrisoft**
axelogo

Using axe-matchers gem to integrate accessibility tests into a set of new cucumber features.

The tests which are run against the pages are listed here:
https://github.com/dequelabs/axe-core/blob/master/doc/rule-descriptions.md

Altered steps which select radio buttons on appointment summary form page to ensure they work with features with/without Javascript

### CODE REVIEW

File: features/accessibility.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
+@javascript
+Feature: Accessibility of pages
+  As a customer I want Pension Wise pages to be fully accessible
+  so I can navigate the website regardless of my disability
```
Comment (benbarnett) [inline features/accessibility.feature]: `regardless of my access needs` maybe reads a little more clearly?

File: features/accessibility.feature [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  @vcr
+  Scenario: Location page is accessible
+    Given I have searched for appointment locations near to a valid postcode
+    When I drill down into a specific search result
+    Then the page should be accessible
+
+  Scenario: Appointment summary page is not accessible until errors have been fixed
+    When I visit the digital appointment summary generator
+    And I select 55 or over as my age range
```
Comment (benbarnett) [inline features/accessibility.feature]: We could do `And I select any age range` and add a step that'll do a `.sample` between the two options maybe?

Comment (benbarnett) [inline features/accessibility.feature]: Up for discussion, not sure how much value this'll give in the scheme of things

File: features/accessibility.feature [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    And I view the appointment summary
+    Then the page should be accessible
+
+  @booking_locations
+  Scenario: Slot picker for face to face booking is not accessible while we develop our own slot picker
+    Given a location is enabled for online booking
+    When I browse for the location
+    And I opt to book online
+    Then the page should not be accessible
```
Comment (benbarnett) [inline features/accessibility.feature]: Was there another way of wording this that better described our situation? I think it was just an alias for this one.

**Comment by benbarnett** [comment]
Loving this so far

File: features/accessibility.feature (hunk 1 above)
Comment (Guntrisoft) [inline features/accessibility.feature]: Done

File: features/accessibility.feature (hunk 2 above)
Comment (Guntrisoft) [inline features/accessibility.feature]: I'm kinda questioning the actual feature specs to be honest - as i'm piecing together some steps in order to get to the page I want to test for accessibility.

Maybe I need something a lot simpler as a feature spec, which then follows some steps in the background.

File: features/accessibility.feature (hunk 3 above)
Comment (Guntrisoft) [inline features/accessibility.feature]: I'm struggling to find what we saw yesterday in the docs!

Comment (Guntrisoft) [inline features/accessibility.feature]: I've altered so it excludes the slot picker but tests the rest of the page

File: features/accessibility.feature (hunk 2 above)
Comment (Guntrisoft) [inline features/accessibility.feature]: @benbarnett I've decided to simplify the feature specs, and keep the details of how I get to the particular page to the step definition - let me know what you think

**Pull request merged**